src/control/server/harness.go

```diff
@@ -188,9 +235,13 @@ func (h *EngineHarness) readyRanks() []system.Rank {
 	defer h.RUnlock()
 
 	ranks := make([]system.Rank, 0)
-	for _, ei := range h.instances {
-		if ei.hasSuperblock() && ei.isReady() {
-			ranks = append(ranks, *ei.getSuperblock().Rank)
+	for idx, ei := range h.instances {
+		if ei.IsReady() {
+			rank, err := ei.GetRank()
+			if err != nil {
+				h.log.Errorf("instance %d: no rank (%s)", idx, err)
+			}
+			ranks = append(ranks, rank)
```

Comment:
If err != nil, should we still be adding the rank to the list?

Reply:
Oops, thanks. Will fix.